### ISSUE TITLE
Passing a block to super -- versions >= 1.5.0

### ISSUE THREAD

**Issue by LeeXGreen** [troessner/reek#379]
I think I have found an issue, starting with version 1.5.0, where code that passes a block to super cannot be properly parsed.

sample Ruby code:

```
class Thing
  def foo
    super { bars.each(&:frobnicate!) }
  end
end
```

This is the error that comes up:

/Users/leeg/.rvm/gems/ruby-2.1.3/gems/reek-1.5.0/lib/reek/smells/nested_iterators.rb:79:in `ignored_iterator?': undefined method `method_name' for (zsuper):#<Class:0x007fa43da72568> (NoMethodError)

I'm attempting to work around this at the moment -- the fact that the code in question doesn't have a `method_name` method just means that we can't ignore it, it seems like we can still check it for quality.

I'm happy to share my workaround if/when I get it working.

**Comment by LeeXGreen** [comment]
Workaround is something like this: https://github.com/LeeXGreen/reek/commit/1208f0808bb1654d8b2e169b75e13652f104d7b9


```diff
-        name = exp.call.method_name.to_s
+        c = exp.call
+        return false unless c.respond_to?(:method_name)
+
+        name = c.method_name.to_s
```

I'm not opening a PR here, because I imagine there's a more nuanced long-term fix to be applied -- but this was the workaround that allowed Reek to run without errors.

**Comment by mvz** [comment]
@LeeXGreen I'll take a look.

**Comment by mvz** [comment]
This was fixed in #365. @troessner, do you think we can add that to a final 1.6 release?

**Comment by troessner** [comment]
I just released 1.6.5 -> closed.
